Diagnostic/Utils/XmlUtil.py

```diff
@@ -26,18 +26,15 @@ def getXmlValue(xml,path,property):
     if element is not None:
         return element.get(property)
 
-def addElement(xml,path,el,selector=[]):
+
+def addElement(xml,path,el,addOnlyOnce=False,selector=[]):
```

Comment:
Won't this break existing code which uses selector (e.g. Utils/ApplicationInsightsUtil.py)? I think you'd need to put addOnlyOnce at the end of the parameters to avoid breaking the signature,